com.unity.multiplayer.mlapi/Runtime/Core/NetworkObject.cs

```diff
@@ -395,10 +393,18 @@ public static void NetworkHide(List<NetworkObject> networkObjects, ulong clientI
 
         private void OnDestroy()
         {
-            if (NetworkManager != null && NetworkManager.SpawnManager != null && NetworkManager.SpawnManager.SpawnedObjects.ContainsKey(NetworkObjectId))
+            if (NetworkManager == null || NetworkManager.SpawnManager == null)
             {
-                NetworkManager.SpawnManager.OnDestroyObject(NetworkObjectId, false);
+                return;
+            }
+
+            if (!NetworkManager.SpawnManager.SpawnedObjects.ContainsKey(NetworkObjectId))
```

Comment:
This code is wrong. A network object can get destroyed when it is not spawned. In this case there shouldn't be anything logged here.

I think what should be done here is remove this if condition with the log here and in the above if condition check for 'IsSpawned'

Reply:
In the original code path, if the network object ID was not in the SpawnedObjects dictionary it would always stop processing, either because of this check, or the check that we're removing in OnDespawnObject(), so is that not the case?

Reply:
Yes it should stop processing. But not log anything. This is regular behavior and not a warning.